sewer/auth.py

```diff
@@ -1,27 +1,35 @@
-import logging
 from .lib import create_logger
+from typing import Any, Dict, Sequence
 
 
-class BaseAuthProvider(object):
-    def __init__(self, auth_type, LOG_LEVEL="INFO"):
-        self.LOG_LEVEL = LOG_LEVEL
-        self.dns_provider_name = self.__class__.__name__
+class ProviderBase:
+    "New-model Provider documentation is in docs/providers.md"
 
-        self.logger = create_logger(__name__, self.LOG_LEVEL)
+    def __init__(self, **kwargs: Any) -> None:
+        # REQUIRED
+        chal_types = kwargs.pop("chal_types")  # throws KeyError if not passed
+        if type(chal_types) not in (list, tuple) or any(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using type() instead of isinstance() for a typecheck.](https://app.codacy.com/manual/komuW/sewer/pullRequest?prid=5574810)